Source/AdUnits/Native/NativeAd.swift

```diff
@@ -0,0 +1,320 @@
+/*   Copyright 2019-2020 Prebid.org, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import UIKit
+
+@objcMembers public class NativeAd: NSObject, CacheExpiryDelegate {
+    
+    private static let nativeAdIABShouldBeViewableForTrackingDuration = 1.0
+    private static let nativeAdCheckViewabilityForTrackingFrequency = 0.25
+    
+    public private(set) var title:String?
+    public private(set) var text:String?
+    public private(set) var iconUrl:String?
+    public private(set) var imageUrl:String?
+    public private(set) var callToAction:String?
+    public private(set) var sponsoredBy:String?
+    
+    public weak var delegate: NativeAdEventDelegate?
+    
+    //NativeAd Expire
+    private var expired = false
+    //Impression Tracker
+    private var targetViewabilityValue = 0
+    private var viewabilityTimer:Timer?
+    private var viewabilityValue = 0
+    private var impressionHasBeenTracked = false
+    private var viewForTracking:UIView?
+    private var impTrackers = [String]()
+    //Click Handling
+    private var gestureRecognizerRecords = [NativeAdGestureRecognizerRecord]()
+    private var clickUrl:String?
+
+    
+    public static func  create(cacheId: String)->NativeAd? {
+        
+        guard let content = CacheManager.shared.get(cacheId: cacheId), let response = Utils.shared.getDictionaryFromString(content), let adm = response["adm"] as? String, let data = adm.data(using: .utf8) else {
+            Log.error("Invalid native contents")
+            return nil
+        }
+        do {
+            guard let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] else {
+                Log.error("Invalid native json")
+                return nil
+            }
+            let ad:NativeAd = NativeAd()
+            //assets
+            if let assets = json["assets"] as? [AnyObject] {
+                for adObject in assets {
+                    if let adObject = adObject as? [String : AnyObject]{
+                        //title
+                        if let title = adObject["title"], let text = title["text"] as? String {
+                            ad.title = text;
+                        }
+                        //description
+                        if let id = adObject["id"] as? Int, id == 3, let description = adObject["data"], let text = description["value"] as? String {
```

Comment:
Hi @avermaiOS 

Could you clarify, please, why did you use the "id" field to determine the object type?